runtime/reconcilers/model.go

```diff
@@ -1721,6 +1799,18 @@ func (r *ModelReconciler) resolveTrigger(ctx context.Context, self *runtimev1.Re
 			return nil, fmt.Errorf("change_mode=patch can only be used with incremental models")
 		}
 
+		// if the catalog has no prior state we take explicit action from user to avoid dropping existing data.
+		if !prevResultExists {
+			if !model.Spec.Trigger && !model.Spec.TriggerFull {
```

Comment:
Maybe check `!model.Spec.TriggerPartitions` here just for completeness? Even though I don't think it would be possible to set it in practice when partitions haven't been synced yet.